app/controllers/concerns/hal_actions.rb

```diff
@@ -59,11 +59,10 @@ def zoom_param
     }.call
   end
 
-
   def index_cache_path
     index_resources = self.try(:resources_base) || resources
     PagedCollection.new(
-      with_params(self.class.resources_params, index_resources),
+      decorate_query(index_resources),
       request,
       item_class: self.class.resource_class,
       item_decorator: self.class.resource_representer
```

Comment:
definitely not something to fix in this PR, but just noticing we can likely optimize away lots of object creation on a cache miss here by initializing only one PagedCollection (i.e. if `resources_base` is unavailable, short circuit to a memoized call to `index_collection`). Wanted to get it written down somewhere